### PR TITLE
Improve the issue templates and pull request template.

### PR DESCRIPTION
**Pull request by guoqiangqi**
Provide better issue templates and pull request template.

### CODE REVIEW

**Comment by esc** [comment]
@guoqiangqi thank you for your suggestion. We will have to discuss this tomorrow in the developer meeting, but personally I am +1 on this. 

Also, could you maybe add a section about checking via Gitter first? if unsure if this is a bug or not? Since the requested answers in the new template are somewhat more involved and I'd like to largely prevent want someone to go through all of the fact gathering to report what then turns out to be a user-error or similar.

**Comment by guoqiangqi** [comment]
> @guoqiangqi thank you for your suggestion. We will have to discuss this tomorrow in the developer meeting, but personally I am +1 on this.
> 
> Also, could you maybe add a section about checking via Gitter first? if unsure if this is a bug or not? Since the requested answers in the new template are somewhat more involved and I'd like to largely prevent want someone to go through all of the fact gathering to report what then turns out to be a user-error or similar.

@esc Sorry about the delay, i have updated this part of code.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -33,3 +33,12 @@ Here's some guidelines to help the review process go smoothly.
 Many thanks in advance for your cooperation!
 
 -->
+
+## Reference issue
```
Comment (esc) [inline .github/PULL_REQUEST_TEMPLATE.md:37]: This is to keep!

**Comment by esc** [comment]
@guoqiangqi thank you again for submitting this improvement! We had a heated conversation about this in the developer meeting yesterday and it turns out, we had refactored the pull-request template to make it simpler! We do see your intention to improve the Numba pull-request and issue-reporting -- but we feel this may be asking a bit too much from our users. I think the main questions that need to be answered are:

* What did you try?
* What did you expect?
* What happened instead?

And then have a runnable (including imports, the code in question and a line or lines to execute it) script to reproduce the bug. Anything else, we can always ask about or request the output of `numba -s`. We do have a first-port-of-call rota in place 
during weekdays so our issue response times are reasonably short and we know what else to ask for. 

Anyhow , the part about "reference an existing issue, if it exists" was well received and will be merged, so that's good! So, I think @guoqiangqi you may modify the PR to include the suggestion about adding a link to a PR and we will merge that!

Thank you again for you efforts on this front and your help to improve Numba!

**Comment by guoqiangqi** [comment]
@esc Agree with you, we should not make the templates complex , it may make issue reporting more complicated. New code has been uploaded.

**Review by esc — APPROVED** [review]
LGTM, thank you!

**Comment by stuartarchibald** [comment]
@guoqiangqi Congratulations on your first contribution to Numba!

@esc thanks for reviewing.

**Pull request merged**